src/components/ConsumableExerciseCard.tsx

```diff
@@ -13,95 +15,142 @@ import {
-import { updateConsumableExercise } from '../utils/db/consumableexercises';
+import { getConsumableExercises, updateConsumableExercise } from '../utils/db/consumableexercises';
+import PastWorkoutPerformance from './PastWorkoutPerformance';
+import { formatDateTime } from '../utils/parsing';
 
 /**
  * @param exercise inputs value to a card
  * @param getUserRecordedSets returns array of user input RecordedValue or undefined if user fails to fil out all values
  */
 export interface ConsumableExerciseCardProps {
-  exerciseName: string;
-  consumableWorkoutId: string;
-}
 
-const ConsumableExerciseCard: React.FC<ConsumableExerciseCardProps> = ({
-  exerciseName,
-  consumableWorkoutId,
-}) => {
-  const { consumableExercise } = useConsumableExercise(exerciseName, consumableWorkoutId);
-  const [exercise, setExercise] = useState<Partial<ConsumableExercise>>(consumableExercise);
-  const [rerender, setRerender] = useState(false);
-  const [exerciseItems, setExerciseItems] = useState<JSX.Element[]>([]);
-
-  useEffect(() => {
-    setExercise(consumableExercise);
-  }, [consumableExercise]);
+    exerciseName: string;
+    consumableWorkoutId: string;
+    userId: string;
+}
 
-  useEffect(() => {
-    let numWarmups = 0;
-    let numWorking = 0;
-    setExerciseItems(
-      exercise.exerciseItems?.map((item: ConsumableExerciseItem, index: number) => {
-        const { ref, data } = item;
-        if ('warmup' in ref && 'reps' in ref && data) {
-          return (
-            <SetBlock
-              key={index}
-              index={index}
-              setNumber={'warmup' in ref && ref.warmup ? ++numWarmups : ++numWorking}
-              setRef={ref}
-              reps={data?.reps}
-              weight={data?.weight}
-              bodyweight={data?.bodyweight}
-              onChange={updateExerciseCardItem}
-            />
-          );
-        } else if ('minutes' in ref && 'seconds' in ref) {
-          return <RestBlock key={index} minutes={ref.minutes} seconds={ref.seconds} />;
-        } else {
-          return <NoteBlock key={index} note={'text' in ref ? ref.text : ''} />;
+const ConsumableExerciseCard: React.FC<ConsumableExerciseCardProps> = ({exerciseName, consumableWorkoutId, userId}) => {
+    const { consumableExercise } = useConsumableExercise(exerciseName, consumableWorkoutId);
+    const [exercise, setExercise] = useState<Partial<ConsumableExercise>>(consumableExercise);
+    const [rerender, setRerender] = useState(false);
+    const [exerciseItems, setExerciseItems] = useState<JSX.Element[]>([]);
+    const [pastExerciseVisible, setpastExerciseVisible] = useState(false);
+    const [pastExericises, setPastExericises] = useState<ConsumableExercise[]>()
+    const [index, setIndex] = useState(0);
+    const [closePastPerformance,setClosePastPerformance] = useState(false);
+    const getPastConsumableExercises = async () =>{
+        try{
+            const data = await getConsumableExercises({userId,exerciseName})

```

Comment:
We don't want to show the current exercise as one of the past exercise options. However, this will be solved when we switch to a local-state version of doing workouts, which are only uploaded to the database when its completed.

Reply:
so don't worry about this rn